### PR TITLE
Add clear error message when sdk can't fetch from keyring

### LINKED ISSUE #75: Keyring might not return the store password
Keyring may not be working on some OS (seen on Debian). 

One encountered error is that it is failing silently to fetch the password. It leads then to an authentication error, this is misleading for the user.

To solve this, we could:
- display a proper error message when keyring fails to fetch the password (Fetching password error: please check your keyring installation)
- update the documentation for Debian (requires https://packages.debian.org/fr/stretch/debian-keyring and https://pypi.org/project/keyrings.alt/) (optional)

### PR DESCRIPTION
**Pull request by AlexandrePicosson**
resolve #75 

The documentation part of this issue is covered by https://github.com/SubstraFoundation/substra/pull/110.

### CODE REVIEW

File: substra/sdk/rest_client.py [hunk 1]
```diff
@@ -89,9 +89,16 @@ def set_config(self, config, profile_name='default'):
             raise exceptions.BadConfiguration('Your configuration is outdated, please update it.')
 
         username = config['auth']['username']
+        password = keyring.get_password(profile_name, username)
+
+        if password is None:
+            raise exceptions.KeyringException(
+                Exception("Fetching password error: Check your keyring installation")
```
Comment (samlesu) [inline substra/sdk/rest_client.py]:
```suggestion
               "Fetching password error: Check your keyring installation"
```

Comment (samlesu) [reply substra/sdk/rest_client.py]: Is there a reason for creating an exception inside an exception?

Comment (AlexandrePicosson) [reply substra/sdk/rest_client.py]: No, i'm just not familiar with the exceptions in python and thought it was the right way to do it

File: substra/sdk/rest_client.py [hunk 2]
```diff
@@ -89,9 +89,16 @@ def set_config(self, config, profile_name='default'):
             raise exceptions.BadConfiguration('Your configuration is outdated, please update it.')
 
         username = config['auth']['username']
+        password = keyring.get_password(profile_name, username)
+
+        if password is None:
+            raise exceptions.KeyringException(
+                'Fetching password error: Check your keyring installation'
+            )
```
Comment (natct10) [inline substra/sdk/rest_client.py]: :+1:

**Pull request merged**